apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -106,9 +107,9 @@
     private final long createTime;
 
     /**
-     * profiling status
+     * profiling data
      */
-    private volatile boolean profiling;
+    private volatile ThreadProfiler profiler;
```

Comment:
I don't think we should open this.

Reply:
Maybe I need to change it to `ProfilingStatus` because we need the status to check the current context is profiling or ready.

Reply:
What do you mean ready? The parameter collection only happens after the profile activated.

Reply:
The profile has `minDurationThreshold` to explain only start profiling when bigger this parameter, before this time, it was ready status. I think on the ready status, we don't need to start the HTTP parameter happens.

Reply:
I think we have a naming issue here. The current `profiling` should rename to `pendingProfile`, and we need a real `profiling` to indicate the thread dump started status.

Reply:
Or you should refactor this as an enum, having value, NONE, PENDING and PROFILING. Also, this enum should have a method called `isBeingWatched`. PENDING and PROFILING should return yes.